go/processor/floria/processor.go

```diff
@@ -70,31 +73,74 @@ func (p *processor) Run(
 		// Create new contract
 	} else {
 		// Call existing contract
-		result, err = call(p.interpreter, transaction, context)
+		result, err = call(p.interpreter, transaction, context, gas)
 		if err != nil {
-			return errorReceipt, nil
+			return tosca.Receipt{GasUsed: transaction.GasLimit}, nil
 		}
 	}
 
+	gasUsed := gasUsed(transaction, result.GasLeft)
+
 	return tosca.Receipt{
 		Success:         result.Success,
-		GasUsed:         transaction.GasLimit,
+		GasUsed:         gasUsed,
 		ContractAddress: nil,
 		Output:          result.Output,
 		Logs:            nil,
 	}, nil
 }
 
+func gasUsed(transaction tosca.Transaction, gasLeft tosca.Gas) tosca.Gas {
+	if transaction.Sender != (tosca.Address{}) {
+		gasLeft -= gasLeft / 10
+	}
+
+	return transaction.GasLimit - tosca.Gas(gasLeft)
+}
+
+// No overflow check for the gas computation is required although it is performed in the
+// opera version. The overflow check would be triggered in a worst case with an input
+// greater than 2^64 / 16 - 53000 = ~10^18, which is not possible with real world hardware
```

Comment:
not possible because the code would have to be longer than 10^18 bytes of ram or why ?

Reply:
also, this code looks like a comment for the operations in line 121, not function documentation